libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -705,6 +706,13 @@ bool AP_Arming::system_checks(bool report)
             check_failed(ARMING_CHECK_SYSTEM, report, "Terrain out of memory");
             return false;
         }
+#endif
+#ifdef ENABLE_SCRIPTING
+        const AP_Scripting *scripting = AP_Scripting::get_singleton();
+        if ((scripting != nullptr) && scripting->init_failed()) {
```

Comment:
I'd recommend removing the changes to `enabled()` and then adding a check here if scripting was enabled. It's quite conceivable you might want to just not use scripting, and just set the enable flag to 0, which wouldn't require a reboot to stop scripts. (The other way would ofc). The main target is just to make it easier to get around this check if you don't need scripts without requiring a reboot.

Reply:
As requested, I've added a check here that scripting was enabled, thanks